packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  SchemaObject,
+  ReferenceObject,
+  isReferenceObject,
+} from '@loopback/openapi-v3-types';
+import {Validator} from './validator';
+import * as HttpErrors from 'http-errors';
+import * as debugModule from 'debug';
+
+const debug = debugModule('loopback:rest:coercion');
+
+/**
+ * Coerce the http raw data to a JavaScript type data of a parameter
+ * according to its OpenAPI schema specification.
+ *
+ * @param data The raw data get from http request
+ * @param schema The parameter's schema defined in OpenAPI specification
+ */
+export function coerceParameter(
+  data: string,
+  schema?: SchemaObject | ReferenceObject,
+) {
+  if (!schema || isReferenceObject(schema)) {
+    debug(
+      'The parameter with schema %s is not coerced since schema' +
+        'dereferrence is not supported yet.',
+      schema,
+    );
+    return data;
+  }
+  let coercedResult;
+  coercedResult = data;
+  const OAIType = getOAIPrimitiveType(schema.type, schema.format);
+  const validator = new Validator({schema});
+
+  switch (OAIType) {
+    case 'byte':
+      coercedResult = Buffer.from(data, 'base64');
+      break;
+    case 'date':
+      coercedResult = new Date(data);
+      break;
+    case 'float':
+    case 'double':
+      coercedResult = parseFloat(data);
+      break;
+    case 'number':
+      validator.validateParamBeforeCoercion('number', data);
+      coercedResult = data ? Number(data) : undefined;
+      validator.validateParamAfterCoercion('number', coercedResult);
+      break;
+    case 'long':
+      coercedResult = Number(data);
+      break;
+    case 'integer':
+      coercedResult = parseInt(data);
+      break;
+    case 'boolean':
+      coercedResult = isTrue(data) ? true : isFalse(data) ? false : undefined;
+    case 'string':
+    case 'password':
+    // serialize will be supported in next PR
+    case 'serialize':
+      break;
+    case 'unknownType':
+    default:
+      throw new HttpErrors.NotImplemented(
+        `Type ${schema.type} with format ${
+          schema.format
+        } is not a valid OpenAPI schema`,
+      );
+  }
+  return coercedResult;
+}
+
+/**
+ * A set of truthy values. A data in this set will be coerced to `true`.
+ *
+ * @param data The raw data get from http request
+ * @returns The corresponding coerced boolean type
+ */
+function isTrue(data: string): boolean {
+  const isTrueSet = ['true', '1'];
+  return isTrueSet.includes(data);
```

Comment:
We should promote ``isTrueSet` outside of the function or simply use `['true', '1'].includes(data)`.